Age of War 3D/Assets/Scripts/Game Controller/AIController.cs

```diff
@@ -9,32 +9,28 @@ public class AIController : BaseGameController
     protected override void Start()
     {
         base.Start();
-        StartCoroutine(SpawnUnit());
-
         //nech brain vie ake lajny existuju
         var lines = new List<Line>();
         for (int i = 0; i < lineController.GetNumberOfActiveLines(); i++)
         {
             lines.Add(lineController.GetLineByIndex(i));
         }
-        aiBrain.Initialize(lines);
+        aiBrain.Initialize(lines, goldController, outpost);
+        StartCoroutine(SpawnUnit());
     }
 
     private IEnumerator SpawnUnit()
     {
         while (true)
         {
-            int activeLines = lineController.GetNumberOfActiveLines();
-
-            //max is exclusive
-            int randomLineIndex = Random.Range(0, activeLines);
-            int randomUnitIndex = Random.Range(0, unitSpawner.NumberOfDifferentUnits - 1);
-
-            _activeLine = lineController.GetLineByIndex(randomLineIndex);
-
-            SpawnUnit(randomUnitIndex);
-
-            yield return new WaitForSeconds(1.7f);
+            var decisions = aiBrain.getNextDecisions();
+
+            foreach (var decision in decisions)
+            {
+                _activeLine = lineController.GetLineByIndex(decision._lineID);
+                SpawnUnit((int)decision._unitType);
+            }
+            yield return new WaitForSeconds(10f);
```

Comment:
Tuto to vyzera ako keby to najskor cakalo a az v dalsom cykle Spawnlo ale chcelo by to sa na to pozriet blizsie

Reply:
možno by bolo fajn checknúť čo tam ide za hodnota prvý krát, lebo vo vnútri UnitSpawneru ak je index mimo povolených tak iba vracia 0 a nič nespraví